### ISSUE TITLE
KendoUI script error after upgrade to latest dev branch

### ISSUE THREAD

**Issue by MRSheikh** [OrchardCMS/OrchardCore#9394]
I developed some modules on Kendo UI framework and that works fine. after upgrade to the latest dev branch, this error occurred on my page:
`Uncaught TypeError: jQuery(...).kendoLoader is not a function`
I think this is because ResorceManagement class moved `<script src="/OrchardCore.Resources/Scripts/jquery.js"></script>` to the end of page.
how can I set `OrchardCore.Resources` scripts inserted in `<head>` tag??

**Comment by jtkech** [comment]
If your resource is defined through a resource manifest

The script tag helper has a new `Position` attribute that you could set to `Last` so that your resource will be registered after.

**Comment by MRSheikh** [comment]
> If your resource is defined through a resource manifest
> 
> There is a new `Position` property that you could set to `Last` so that your resource will be registered after.
> 
> Hmm, but you can also just make you resource depend on jQuery, or use the tag helper `depends-on` attribute.

yes I know that but how to move build-in ` OrchardCore.Resources/jquery.js ` in to head or remove that ??

**Comment by weirdyang** [comment]
https://docs.orchardcore.net/en/dev/docs/reference/modules/Resources/

You can do it programmatically or use the liquid positioning at: head

**Comment by MichaelPetrinolis** [comment]
@jtkech here is what I encountered while playing with Kendo in custom admin Theme. In the Layout.cshtml I add Kendo js/css as resources which has been registered in a custom module. I have set Kendo's jQuery version as Dependency for kendo js. I add kendo at=Head but jQuery is loaded at=Foot. I add a script tag explicitly for jQuery at=Head, the same thing. That's because the jquery-cookie is register at=Foot and depends on jQuery. The resource manager for some reason moves jQuery at=Foot and it breaks kendo. 
I explicitly set jquery-cookie at=Head and works. 

I did not had time to investigate, but we should add some tests for that scenario.

**Comment by jtkech** [comment]
If A at Foot depends on B at Head, B should stay at Head

I found the issue, i tried it it works, will do a PR this night ;)

**Comment by MichaelPetrinolis** [comment]
@jtkech I made a PR with a fix

**Comment by MichaelPetrinolis** [comment]
Sorry, didn't see the comment in the PR

**Comment by Skrypt** [comment]
@jtkech 

A (at head) depends on B (at foot) = B should be at head before A.

But what if I have C (at foot) which also depends on B (at foot) and that it needs to be (at foot) ...
I think we are running in circle trying to find how we should handle dependencies when they should be handle with defered scripts. Of course, these leaves out inline scripts but there is surely a workaround for these too as they could be set as defered too.

**Comment by jtkech** [comment]
> A (at head) depends on B (at foot) = B should be at head before A.

Yes this is how it is intended to work when we combine a location with another one

        public RequireSettings AtLocation(ResourceLocation location)
        {
            // if head is specified it takes precedence since it's safer than foot
            Location = (ResourceLocation)Math.Max((int)Location, (int)location);
            return this;
        }

But If A at Foot depends on B at Head, B is intended to stay at Head

There is just a missing line somewhere before combining dependency settings

        public RequireSettings NewAndCombine(RequireSettings other)
        {
            return new RequireSettings
            {
                Name = Name,
                Type = Type,
                Location = Location, <= The missing line
                Position = Position
            }
                .Combine(other)
                ;
        }

**Comment by Skrypt** [comment]
Ok, I find this kind of odd to be honest. This means that in any view I could change where JQuery is declared by declaring a script that has a dependency on it and that I want to render at foot. Which could potentially break other global scripts which would require it to be at foot?

**Comment by MRSheikh** [comment]
thanks. I found that some `<script>` on my pages `depends-on='jQuery'` and so that, Orchardcore added `jquery.js` automatically on my pages. I removed all `depends-on='jQuery'` and added `jquery.min.js` manually and then Kendo-Ui components work fine !!

**I think everyone who wants to use  Kendo-Ui components should be aware of these notes:**
1- just only one `jquery.js` should be added to the page.
2- `jquery.js`, `kendo.all.min.js`, `kendo.aspnetmvc.min.js`  should be added to the `<head>` section of the page respectively.
